app/models/media.rb

```diff
@@ -103,4 +125,40 @@ def set_type
   def set_uuid
     self.update_column(:uuid, self.id)
   end
+
+  def set_original_claim_hash
+    self.original_claim_hash = Digest::MD5.hexdigest(original_claim) unless self.original_claim.blank?
+  end
+
+  def self.create_uploaded_file_media_from_original_claim(media_type, url, ext)
+    klass = media_type.constantize
+    file = download_file(url, ext)
```

Comment:
Please move this to inside the `else` block below - the idea of this method is to only download the file if the hash doesn't exist.